community/cypher/cypher/src/test/scala/org/neo4j/cypher/SemanticErrorAcceptanceTest.scala

```diff
@@ -666,6 +666,18 @@ class SemanticErrorAcceptanceTest extends ExecutionEngineFunSuite {
       "A single relationship type must be specified for MERGE (line 1, column 9 (offset: 8))")
   }
 
+  test("give a nice error message when missing crs in cartesian point") {
+    executeAndEnsureError("RETURN point({x: 2.3, y: 4.5}) as point",
+                          "A cartesian point must contain a 'crs' (coordinate reference system) (line 1, column 14 (offset: 13))")
```

Comment:
I changed my mind. I think it would be better to keep your error message, and make the xy/latlng the primary criteria for geo/cartesian, which is what I assume you thought was the case. I think your approach is better, but it requires that we no longer assume WGS84 default for missing crs, but have two defaults, one for xy and one for latlng. That change does not need to be made as part of this PR either, as it is just a small refinement to the case of missing crs.